src/txn/txn_timestamp.c

```diff
@@ -879,6 +879,49 @@ __wt_txn_set_read_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t read_ts)
     return (0);
 }
 
+/*
+ * __txn_set_timestamp_internal --
+ *     Does the "heavy lifting" for the outside-facing txn_set_timestamp functions.
+ */
+static int
+__txn_set_timestamp_internal(
+  WT_SESSION_IMPL *session, uint64_t commit, uint64_t durable, uint64_t prepare, uint64_t read)
+{
+    WT_CONNECTION_IMPL *conn;
+    bool set_ts;
+
+    conn = S2C(session);
+    set_ts =
+      commit != WT_TS_NONE || durable != WT_TS_NONE || prepare != WT_TS_NONE || read != WT_TS_NONE;
```

Comment:
It's also curious that GitHub thinks "read" is a function (or something) by highlighting it blue, I wonder if the parameters should have `_ts` appended to them?

Reply:
Regarding your first point - it's not quite the same, since you skip publishing the durable (and commit) timestamp and the logging step. I intended to match the old behaviour here, but if you feel this change is in scope for this ticket then I'll work on it. There are a couple of comments along these lines, so let's keep the conversation here in one thread?

Fair point on the `_ts` name, Hari mentioned that earlier but this is the first concrete instance of some tooling getting confused, so I'll fix it.

Reply:
Fixed the names in bc79889